src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -1136,6 +1137,39 @@ private BidderResponse rejectBidderResponseOrProceed(HookStageExecutionResult<Bi
                 .map(seatBid -> BidderResponse.of(bidderName, seatBid, responseTime(startTime)));
     }
 
+    private List<BidderResponse> dropZeroNonDealBids(List<BidderResponse> bidderResponses, List<String> debugWarnings) {
+
+        return bidderResponses.stream()
+                .map(bidderResponse -> removeBidsWithInvalidBids(bidderResponse, debugWarnings))
+                .collect(Collectors.toList());
+    }
+
+    private BidderResponse removeBidsWithInvalidBids(BidderResponse bidderResponse, List<String> debugWarnings) {
```

Comment:
Maybe rename to smth like filterBids/removeInvalidPriceBids both this and top methods? Cuz this name is not self explanatory, and top function name hides the fact that you actually remove not only non deal zero bids, but all bids with invalid price.